bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Cursor.java

```diff
@@ -646,6 +629,13 @@ private static class ImageDataProviderCursorHandleProvider extends HotspotAwareC
 
 	public ImageDataProviderCursorHandleProvider(ImageDataProvider provider, int hotspotX, int hotspotY) {
 		super(hotspotX, hotspotY);
+		ImageData source = provider.getImageData(DEFAULT_ZOOM);
+		if (source == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
+		/* Check the hotspots */
+		if (hotspotX >= source.width || hotspotX < 0 ||
+			hotspotY >= source.height || hotspotY < 0) {
+			SWT.error(SWT.ERROR_INVALID_ARGUMENT);
+		}
```

Comment:
This check (and others in other constructors) now makes the same one inside `setupCursorFromImageData` obsolete as they are already checked here in the provider constructors, so the replicated checks should be removed there.